aio/aio-proxy/aio_proxy/response/build_response.py

```diff
@@ -5,31 +5,33 @@
 from aio_proxy.decorators.http_exception import http_exception_handler
 from aio_proxy.response.format_response import format_response
 from aio_proxy.response.format_search_results import format_search_results
-from aio_proxy.search.index import ElasticsearchSireneIndex
+from aio_proxy.search.es_search_runner import ElasticSearchRunner
 from aiohttp import web
 from sentry_sdk import capture_exception, push_scope
 
 
 @http_exception_handler
 def api_response(
-    request, extract_function: Callable, search_function: Callable
+    request,
+    extract_function: Callable,
+    search_type: str,
 ) -> dict[str, int]:
     """Create and format API response.
 
     Args:
         request: HTTP request.
         extract_function (Callable): function used to extract parameters.
-        search_function (Callable): function used for search .
+        search_type: type of search (text or geo).
     Returns:
         response in json format (results, total_results, page, per_page,
         total_pages)
     """
     try:
         search_params = extract_function(request)
```

Comment:
Open suggestion : 
maybe merge `format_search_results` into `ElasticSearchRunner` 

Technically not related to elastic so you might need a more generic name though.

This would allow very simple code in api_response : 
```
search_params = extract_params(request, search_type)
search_query = QueryBuilder(search_params, search_type)
results = search_query.run()
results_formatted = format_search_results(results, search_params)
```
where results contains `results_formatted, total_results, execution_time`

Reply:
I think it's outside the scope of this particular PR, but will be addressed in the following refacto PR.